src/Elmish/Component.purs

```diff
@@ -307,13 +307,13 @@ bindComponent cmpt stateStrategy = \def -> -- Explicit lambda to make sure `def`
           sequence_ =<< getSubscriptions component
           setSubscriptions [] component
 
-        subscriptionsField = "__subscriptions"
-        getSubscriptions = getField @(Array (Effect Unit)) subscriptionsField >>> map (fromMaybe [])
-        setSubscriptions = setField @(Array (Effect Unit)) subscriptionsField
+        subscriptionsField = Field @"__subscriptions" @(Array (Effect Unit))
+        getSubscriptions = getField subscriptionsField >>> map (fromMaybe [])
+        setSubscriptions = setField subscriptionsField
 
-        unmountedField = "__unmounted"
-        getUnmounted = getField @Boolean unmountedField >>> map (fromMaybe false)
-        setUnmounted = setField @Boolean unmountedField
+        unmountedField = Field @"__unmounted" @Boolean
```

Comment:
The `Field` instance binds together field name and type, making mistakes less likely.